File: BackEnd/FastApiServer/models/keyword.py
# models/keyword.py
from sqlalchemy import Column, BigInteger, String
from sqlalchemy.orm import relationship
from database import Base

class Keyword(Base):
    __tablename__ = "keyword"

    id = Column(BigInteger, primary_key=True, index=True)
    japanese = Column(String(255), nullable=False)
    news_keyword_mappings = relationship("NewsKeywordMapping", back_populates="keyword")
    news_keyword_histories = relationship("NewsKeywordHistory", back_populates="keyword")

File: BackEnd/SpringBootServer/src/main/java/com/ssafy/backend/domain/news/entity/Keyword.java
package com.ssafy.backend.domain.news.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@Entity
public class Keyword {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "INT UNSIGNED")
    private Long id;

    @Column(nullable = false)
    private String japanese;

    private String korean;

    @OneToMany(mappedBy = "keyword")
    private List<NewsKeywordMapping> newsKeywordMappings = new ArrayList<>();

    @OneToMany(mappedBy = "keyword")
    private List<NewsKeywordHistory> newsKeywordHistories = new ArrayList<>();
}
